File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/application/port/outbound/bike/BikeSubscriptionRepository.java
package io.miragon.example.application.port.outbound.bike;

import io.miragon.example.domain.bike.BikeSubscription;
import io.miragon.example.domain.bike.BikeSubscriptionId;

public interface BikeSubscriptionRepository {
    BikeSubscription save(BikeSubscription subscription);
    BikeSubscription find(BikeSubscriptionId id);
    BikeSubscription search(BikeSubscriptionId id);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/application/port/outbound/bike/BikeSubscriptionRepository.kt
package io.miragon.example.application.port.outbound.bike

import io.miragon.example.domain.bike.BikeSubscription
import io.miragon.example.domain.bike.BikeSubscriptionId

interface BikeSubscriptionRepository {
    fun save(subscription: BikeSubscription): BikeSubscription
    fun find(id: BikeSubscriptionId): BikeSubscription
    fun search(id: BikeSubscriptionId): BikeSubscription?
}
